### PR TITLE
Updates in Organizers

### PR DESCRIPTION
**Pull request by mitogh**
- Remove usage of meta to store temp data
- Sync organizer into the store
- Sync new / classic editor with the organizers from stores
- Remove / Edit organizers as described on requirements
- Update organizer component into HOC

### CODE REVIEW

File: src/modules/data/organizers/index.js [hunk 1]
```diff
@@ -21,6 +23,40 @@ const details = {
 				payload,
 			};
 		},
+		removeDraft( id ) {
+			return {
+				type: 'REMOVE_DRAFT',
```
Comment (paulmskim) [inline src/modules/data/organizers/index.js]: here, and in another place above, we're using strings to denote action ID. this is fine, but at some point I think moving to constants to keep things consistent if we use the same action across different components would be key.

File: src/modules/data/organizers/reducers.js [hunk 2]
```diff
@@ -263,3 +278,27 @@ function editPost( prevState, id, payload ) {
...
+	const { organizer } = block;
+
+	if ( ! organizer ) {
+		return state;
+	}
+
+	apiRequest( {
+		path: `/wp/v2/${ POST_TYPE }/${ organizer }`,
+		method: 'DELETE',
+	} ).then( () => {
```
Comment (paulmskim) [inline src/modules/data/organizers/reducers.js]: here we can also leverage shorthand:
`.then( () => dispatch( STORE_NAME ).clear( id ) );`

File: src/modules/data/organizers/index.js (hunk 1 above)
Comment (mitogh) [reply src/modules/data/organizers/index.js]: I think is a great idea; we can include on that one as well the split of files from stores into his own files for:

- Dispatcher
- Actions
- Selectors

Into his own files to keep store declaration very small and easy to debug.

File: src/modules/blocks/classic-event-details/organizer/actions.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	);
+
+	return (
+		<IconButton
+			className="tribe-editor__btn"
+			label={ __( 'Remove Organizer', 'events-gutenberg' ) }
+			onClick={ onRemoveOrganizer }
+			icon={ icon }
+			aria-expanded={ focus }
+			style={ { position: 'absolute', right: 0, top: '-5px' } }
```
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: I think that's a fair point this came from old code base but is a good time to do a refactor into is own file 👍

File: src/modules/blocks/classic-event-details/organizer/actions.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		<Dashicon icon="no"/>
+	);
+
+	return (
+		<IconButton
+			className="tribe-editor__btn"
+			label={ __( 'Remove Organizer', 'events-gutenberg' ) }
+			onClick={ onRemoveOrganizer }
+			icon={ icon }
+			aria-expanded={ focus }
```
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: This was a property from the old file but I think I didn't pass it into this component, nice catch 👍

File: src/modules/blocks/classic-event-details/organizer/actions.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from 'data/organizers';
+
+const ActionsComponent = ( { isBlock, onRemoveOrganizer = noop } ) => {
+
+	if ( isBlock ) {
```
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: Right! I think as the next part of the ternary operator is multiple lines makes things more hard to read if we do it in a ternary operator. I think having a condition with an early return makes more clear how the component behaves.

File: src/modules/blocks/classic-event-details/organizer/item.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import { Loading } from 'elements';
+import { Actions, Name } from './index';
+
+const Item = ( { id, data = {}, isLoading, onRemoveOrganizer } ) => {
+
+	if ( isLoading ) {
+		return <li><Loading className="tribe-editor__spinner--item" /></li>;
+	}
+
+	return (
```
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/item.js]: I think this one is same as above. I think we should favor readability on this one as inline with a multiple line block is not easy to update / modify / read.

**Comment by mitogh** [comment]
Changes requested has been updated @paulmskim 

Let me know what do you think.

Thank you!

File: src/modules/blocks/classic-event-details/organizer/actions.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (paulmskim) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: this is fair, whatever is the end goal, readability or conciseness.

File: src/modules/blocks/classic-event-details/organizer/actions.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulmskim) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: will you be refactoring this in this pass or another pass?

File: src/modules/blocks/classic-event-details/organizer/actions.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: I will favor `readability` as `conciseness` is great when does not affect `readability`.

File: src/modules/blocks/classic-event-details/organizer/actions.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Import External dependencies
+ */
+import React from 'react';
+import { noop } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import {
```
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: Not sure why the change is not reflected on the diff but has been updated accordingly: 

- https://github.com/moderntribe/events-gutenberg/blob/refactor-selectors/src/modules/blocks/classic-event-details/organizer/actions.js#L16

File: src/modules/blocks/classic-event-details/organizer/actions.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mitogh) [reply src/modules/blocks/classic-event-details/organizer/actions.js]: Yes let me update this file.

**Pull request merged**